araproc/analysis/impulsivity_and_matrices.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.signal import hilbert
+from scipy.stats import linregress
+from scipy.optimize import curve_fit
+from scipy.special import erf
+
+# Define the erf-linear model for curve fitting
+def erf_linear(x, A, B):
+    return (A * erf(x / B) + x) / (A * erf(1 / B) + 1)
+
+# Generate a Gaussian noise waveform (600 ns)
+np.random.seed(42)
+sampling_interval = 0.5  # ns
+time_array = np.arange(0, 600, sampling_interval)
+voltage_array = np.random.normal(0, 1, len(time_array))
+
+# Function to calculate impulsivity-related variables
+def calculate_impulsivity_measures(voltage_array,time_array):
```

Comment:
Lets add our usual documentation about what it does, inputs, and outputs to this function.